src/subclass/widget.rs

```diff
@@ -0,0 +1,249 @@
+                .expect("No parent class impl for \"adjust_baseline_request\"");
+            f(
+                widget.to_glib_none().0,
+                minimum_baseline,
+                natural_baseline,
+            )
+        }
+    }
+
+    fn parent_adjust_size_allocation(&self, widget: &Widget, orientation: Orientation, minimum_size: &mut i32, natural_size: &mut i32, allocated_pos: &mut i32, allocated_size: &mut i32) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            let f = (*parent_class)
+                .adjust_size_allocation
+                .expect("No parent class impl for \"adjust_size_allocation\"");
+            f(
+                widget.to_glib_none().0,
+                orientation.to_glib(),
+                minimum_size,
+                natural_size,
+                allocated_pos,
+                allocated_size,
+            )
+        }
+    }
+
+    fn parent_adjust_size_request(&self, widget: &Widget, orientation: Orientation, minimum_size: &mut i32, natural_size: &mut i32) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            let f = (*parent_class)
+                .adjust_size_request
+                .expect("No parent class impl for \"adjust_size_request\"");
+            f(
+                widget.to_glib_none().0,
+                orientation.to_glib(),
+                minimum_size as *mut i32,
+                natural_size as *mut i32,
+            )
+        }
+    }
+
+    fn parent_button_press_event(&self, widget: &Widget, event: &mut gdk::EventButton) -> bool {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            let f = (*parent_class)
+                .button_press_event
+                .expect("No parent class impl for \"button_press_event\"");
+            f(widget.to_glib_none().0, event.to_glib_none_mut().0) != 0
+        }
+    }
+
+    fn parent_button_release_event(&self, widget: &Widget, event: &mut gdk::EventButton) -> bool {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            let f = (*parent_class)
+                .button_release_event
+                .expect("No parent class impl for \"button_release_event\"");
+            f(widget.to_glib_none().0, event.to_glib_none_mut().0) != 0
+        }
+    }
+}
+
+unsafe impl<T: ObjectSubclass + WidgetImpl> IsSubclassable<T> for WidgetClass {
+    fn override_vfuncs(&mut self) {
+        unsafe extern "C" fn widget_adjust_baseline_allocation<T: ObjectSubclass>(
```

Comment:
I would move these trampolines outside the `override_vfuncs` function. Otherwise you'll end up with 100s lines of trampoline functions declared inside here :) It's going to be hard to read, just move them all to the bottom of the file.